binding/flutter/android/src/main/java/ai/picovoice/flutter/rhino/RhinoPlugin.java

```diff
@@ -11,23 +11,163 @@
+    DELETE
+  }
+
+  private Context flutterContext;
+  private MethodChannel channel;
+  private final Map<String, Rhino> rhinoPool = new HashMap<>();
 
   @Override
-  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {  
+  public void onAttachedToEngine(@NonNull FlutterPluginBinding flutterPluginBinding) {
+    flutterContext = flutterPluginBinding.getApplicationContext();
+    channel = new MethodChannel(flutterPluginBinding.getBinaryMessenger(), "rhino");
+    channel.setMethodCallHandler(this);
+  }
+
+  @Override
+  public void onMethodCall(@NonNull MethodCall call, @NonNull Result result) {
+    Method method;
+    try {
+      method = Method.valueOf(call.method.toUpperCase());
+    } catch (IllegalArgumentException e) {
+      result.error(
+              RhinoRuntimeException.class.getSimpleName(),
+              String.format("Porcupine method '%s' is not a valid function", call.method),
+              null);
+      return;
+    }
+
+    switch (method) {
+      case CREATE:
+        try {
+          String accessKey = call.argument("accessKey");
+          String modelPath = call.argument("modelPath");
+          String contextPath = call.argument("contextPath");
+          Double sensitivity = call.argument("sensitivity");
+          Boolean requireEndpoint = call.argument("requireEndpoint");
+
+          Rhino.Builder rhinoBuilder = new Rhino.Builder()
+                  .setAccessKey(accessKey)
+                  .setModelPath(modelPath)
+                  .setContextPath(contextPath);
+
+          if (sensitivity != null) {
+            rhinoBuilder.setSensitivity(sensitivity.floatValue());
+          }
+
+          if (requireEndpoint != null) {
+            rhinoBuilder.setRequireEndpoint(requireEndpoint);
+          }
+
+          Rhino rhino = rhinoBuilder.build(flutterContext);
+          rhinoPool.put(String.valueOf(System.identityHashCode(rhino)), rhino);
+
+          Map<String, Object> param = new HashMap<>();
+          param.put("handle", String.valueOf(System.identityHashCode(rhino)));
+          param.put("contextInfo", rhino.getContextInformation());
+          param.put("frameLength", rhino.getFrameLength());
+          param.put("sampleRate", rhino.getSampleRate());
+          param.put("version", rhino.getVersion());
+
+          result.success(param);
+        } catch (RhinoException e) {
+          result.error(e.getClass().getSimpleName(), e.getMessage(), null);
+        } catch (Exception e) {
+          result.error(RhinoException.class.getSimpleName(), e.getMessage(), null);
+        }
+        break;
+      case PROCESS:
+        try {
+          String handle = call.argument("handle");
+          ArrayList<Integer> pcmList = call.argument("frame");
+
+          if (!rhinoPool.containsKey(handle)) {
+            result.error(
+                    RhinoInvalidArgumentException.class.getSimpleName(),
+                    "Invalid Porcupine handle provided to native module",
```

Comment:
Porcupine -> Rhino. Also, I think if rhinoPool doesn't contain the handle it's more of an InvalidStateException because it is most likely the user has deleted a rhino and then called process.